### DISCUSSION TITLE
sftp bookmark and private key

### DISCUSSION THREAD

**Discussion by czosnkowy666** [mucommander/mucommander#1015] [Q&A] [answered]
With new version we have improved support of sftp and private key used for authentication - it is great :) 
But how we can save it in bookmark that given connection should use a key instead of password?

**Comment by ahadas** [comment, marked as answer]
@czosnkowy666 the bookmark only holds a URL and is not affected by whether ssh-key or password is used to connect to that URL. the credentials, with either ssh-key or password, are stored separately if you choose them to be persistent.

**Comment by czosnkowy666** [comment]
Yes, I found that even if I don't have password but a key I need to select "store password" option to save ssh key used for connection. All good, Thanks

**Reply by ahadas** [reply]
ah right, need to change `store password` to `store credentials`...

**Reply by ahadas** [reply]
posted https://github.com/mucommander/mucommander/pull/1030

**Comment by 10der** [comment]
hi!

How to force using /Users/myusername/.ssh/config 
i needs to force HostKeyAlgorithms=+ssh-rsa for server, cuz I am a not a server owner (can't change server daemon)

thank you.

**Reply by ahadas** [reply]
@10der looking at the jsch library we use, [it seems like ssh-rsa is used by default](https://github.com/mwiede/jsch/blob/d4c68695f96e3a723e84ef5923a1aa4c3ad3e58c/ChangeLog#L883) but I'm not sure. anyway, to your question, while [it's possible to make jsch use ~/.ssh/config](https://github.com/mwiede/jsch/wiki/Jsch-Configuration), this option doesn't exist in muCommander - feel free to file an issue if you're interested in this

**Reply by 10der** [reply]
Thank you so much.

btw, 
![image](https://github.com/mucommander/mucommander/assets/7056498/734d8f68-6952-48ed-b5a2-ff19e75b0470)
![image](https://github.com/mucommander/mucommander/assets/7056498/87196dd9-5f6a-465e-ae3b-02071d03ca63)

**Comment by 10der** [comment]
mc
![image](https://github.com/mucommander/mucommander/assets/7056498/d2997d76-c4b1-406c-9ddf-465b25ad1372)
![image](https://github.com/mucommander/mucommander/assets/7056498/f6771071-adb7-4e13-9186-4468451c52fa)
